### PR TITLE
feat(suspect-spans): Support filters on span group

### PR DESCRIPTION
**Pull request by Zylphrex**
This adds a filter on the span group in order to narrow down a specific span.
This is using a query parameter instead of being a part of the query string
because the search filter would require an `arrayJoin` in the conditions, and
there's no way to guard other parts of the product (e.g. Discover) from using
this search condition which can result in an unexpected `arrayJoin`.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_spans_performance.py [hunk 1]
```diff
@@ -257,17 +265,29 @@ def query_suspect_span_groups(
...
     if span_ops:
-        builder.add_conditions(
-            [
-                Condition(
-                    builder.resolve_function("array_join(spans_op)"),
-                    Op.IN,
-                    Function("tuple", span_ops),
-                ),
-            ]
+        extra_conditions.append(
```
Comment (k-fish) [inline src/sentry/api/endpoints/organization_events_spans_performance.py:271]: Is there a reason you're collecting into `extra_conditions` first? You can just do `add_conditions([...])` twice, no?

File: tests/snuba/api/endpoints/test_organization_events_spans_performance.py [hunk 2]
```diff
@@ -774,7 +774,6 @@ def test_op_filters(self, mock_raw_snql_query):
                 self.url,
                 data={
                     "project": self.project.id,
-                    "sort": "-sumExclusiveTime",
```
Comment (k-fish) [inline tests/snuba/api/endpoints/test_organization_events_spans_performance.py:777]: Cleaner, nice :clap:

**Review by k-fish — APPROVED** [review]
You should also probably write a test for the `spanGroup` `is_span_id` error.

File: src/sentry/api/endpoints/organization_events_spans_performance.py (hunk 1 above)
Comment (Zylphrex) [reply src/sentry/api/endpoints/organization_events_spans_performance.py:271]: No specific reason

**Pull request merged**